sponsorship_switzerland/models/completion_rules.py

```diff
@@ -79,9 +80,23 @@ def get_from_partner_ref(self, stmts_vals, st_line):
         if 'ref' in st_line:
             ref = st_line['ref']
         res = {}
+
+        partner_ref = ref[9:16]
         partner_obj = self.env['res.partner']
-        partner = partner_obj.search(
-            [('ref', '=', str(int(ref[9:16])))])
+        partner = partner_obj.search([('ref', '=', str(int(partner_ref)))])
+        if not partner:
+            # Some bvr reference have a wrong number of leading zeros,
+            # resulting in the partner reference to be offset.
+            flexible_ref_match = re.search(r'^0{,10}([1-9]\d{6})0',
+                                           ref)
+            if flexible_ref_match:
+                flexible_ref = flexible_ref_match.group(1)
+                if int(flexible_ref) != int(partner_ref):
```

Comment:
Not sure why this test is needed? Wouldn't it work if they were equal?

Reply:
If they are equal it means that we have already tries to resolve flexible_ref and don't need to try again.